spec/lib/bookmarks_bulk_action_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+RSpec.describe BookmarksBulkAction do
+  fab!(:user) { Fabricate(:user, refresh_auto_groups: true) }
+  fab!(:user_2) { Fabricate(:user, refresh_auto_groups: true) }
+  fab!(:bookmark_1) { Fabricate(:bookmark, user: user) }
+  fab!(:bookmark_2) { Fabricate(:bookmark, user: user) }
+
+  describe "delete" do
+    describe "when user can't delete" do
+      it "does NOT delete the bookmarks" do
+        bba = BookmarksBulkAction.new(user_2, [bookmark_1.id, bookmark_2.id], type: "delete")
+        expect { bba.perform! }.to raise_error Discourse::InvalidAccess
+
+        expect(Bookmark.where(id: bookmark_1.id)).to_not be_empty
+        expect(Bookmark.where(id: bookmark_2.id)).to_not be_empty
+      end
+    end
+
+    describe "when user can't edit" do
```

Comment:
This test case is a duplicate of the previous one.

Reply:
Yes, it seems right now the users who can delete are the same as the ones who can edit:
<img width="528" alt="Screenshot 2024-05-06 at 22 13 10" src="https://github.com/discourse/discourse/assets/15239141/4900f81d-0776-4038-881d-de114aa6628e">

I was unsure if adding tests for users who could edit would make sense since they would be the same. However, as you might change this in the future, I duplicated the tests. I'm removing the duplication, and the test can be added when things change.